### PR TITLE
Masonry: Enable two column module on secod batch

### PR DESCRIPTION
**Pull request by manvoid**
### Summary

This PR fixes an issue where the page would freeze trying to calculate the position of a two column module when there are many items that have the height already measured but not the position. This is required to enable the use of a two column module on the first page (with the items comming from the server).

#### Detail

When Masonry is first rendered it measures the heights of all the items at the same time, after that it calculates the positions. This was not a problem for one column items because the process to calculate the positions is very fast, but for two column items this is really hard because it has to create the graph for all the items not limited to the batch size, depending on the amount of items this can freeze the page completely. 

To fix it we can force the batch size when there is a two column module taking into account the items that don't have positions and limiting those to the batch size.

This PR also fixes a bug on the integration test page were the event are added twice.

### CODE REVIEW

File: docs/integration-test-helpers/masonry/MasonryContainer.js
Comment (willsmanley) [inline docs/integration-test-helpers/masonry/MasonryContainer.js:106]: just curious, why do we need to count errors here?

Comment (manvoid) [reply docs/integration-test-helpers/masonry/MasonryContainer.js:106]: Not completely sure but I see that we have two integration test on playwright that check for that variable. `playwright/masonry/null-items.spec.mjs` and `playwright/masonry/handle-update-during-insertions.spec.mjs`.

Comment (willsmanley) [reply docs/integration-test-helpers/masonry/MasonryContainer.js:106]: ah i see thanks. well PR LGTM but i cant approve it.

File: packages/gestalt/src/Masonry.js
Comment (liuyenwei) [inline packages/gestalt/src/Masonry.js:594]: this works since it effectively limits the number of items we use to calculate new positions, but the downside here is that it will take more render cycles to complete, depending on where the two column item is positioned.

example:
assuming 16 items, where the two col module is in position 9 (your example in the test):
- first pass render 6 items
- second pass render 6 additional items
- third pass render 6 additional items (this one has the two column module)
- fourth pass render the remaining 4

alternatively, what if we continue to render/layout all  items but leave it up to the layout on how many items it wants to factor into the graph by limiting the size of this array here: https://github.com/pinterest/gestalt/blob/0a517310803467aaf2bc1f017d88ad38405dee26/packages/gestalt/src/Masonry/defaultTwoColumnModuleLayout.js#L266. This should achieve a similar effect since it limits the number of items that are factored into the graph but allow us to do a full layout sooner

Going off the previous example, we would render all 16 items in one pass but the two column layout would only take X amount of items into account

File: packages/gestalt/src/Masonry/defaultTwoColumnModuleLayout.js
Comment (liuyenwei) [inline packages/gestalt/src/Masonry/defaultTwoColumnModuleLayout.js:302]: this also has the effect of "pushing up" later one column items (again, very dependent on what the incoming items looks like)

assuming we have 24 items (so 4 batches) and the two column item is in the second batch, this would result in batches 1, 3 and 4 being positioned before the second batch.

instead i think what we want to do is split into three batches:
- all items before two column batch
- batch containing two column item
- everything else

first batch would then be `prevOneColumnItems`, second batch would be `batchWithTwoColumnItems` and third batch would be handled already at the end via https://github.com/pinterest/gestalt/pull/3458

File: packages/gestalt/src/Masonry/defaultTwoColumnModuleLayout.js
Comment (liuyenwei) [inline packages/gestalt/src/Masonry/defaultTwoColumnModuleLayout.js:281]: the naming for some of these variables is now pretty confusing/inaccurate imo (e.g. `prevOneColumnItems` here could actually contain two column items)

**Review by liuyenwei — APPROVED** [review]

File: playwright/masonry/two-module.spec.mjs
Comment (liuyenwei) [inline playwright/masonry/two-module.spec.mjs:60]: okay for a followup - ideally we'd update the integration test harness to specify initial items and test the initial page scenario here

Comment (manvoid) [reply playwright/masonry/two-module.spec.mjs:60]: Yeah, my current solution is to set the items with this event and after that toggle the mount of the component off and on and it works but we could find a more straightforwad solution for that.

Comment (manvoid) [reply playwright/masonry/two-module.spec.mjs:60]: I'll create a backlog item for that.

File: packages/gestalt/src/Masonry/defaultTwoColumnModuleLayout.js
Comment (manvoid) [reply packages/gestalt/src/Masonry/defaultTwoColumnModuleLayout.js:306]: We need to add the positions to the cache, this come from the `paintedItemPositions` but that is an array of `$ReadOnlyArray<{ item: T, position: Position }>`, we would have to search for each prev item on that array, adding everything seems easier although I understand your comment. We can double check this on our sync.

File: packages/gestalt/src/Masonry/defaultTwoColumnModuleLayout.js
Comment (manvoid) [reply packages/gestalt/src/Masonry/defaultTwoColumnModuleLayout.js:10]: Yes. I'll create a backlog ticket for this.

File: packages/gestalt/src/Masonry/defaultTwoColumnModuleLayout.js
Comment (manvoid) [reply packages/gestalt/src/Masonry/defaultTwoColumnModuleLayout.js]: As suggested used `prev` instead.

File: packages/gestalt/src/Masonry/defaultTwoColumnModuleLayout.js
Comment (manvoid) [reply packages/gestalt/src/Masonry/defaultTwoColumnModuleLayout.js]: Changed to a more straightforward solution creating only three parts, `pre` before the two column batch, the two column batch and the rest.

**Pull request merged**
